src/lua/init.lua

```diff
@@ -86,81 +111,75 @@ local function load_lua(file)
     return loadfile(file)
 end
 
-local function search_cwd_lib(name)
-    local path = "./?."..soext
-    return package.searchpath(name, path)
-end
-
-local function search_cwd_lua(name)
-    local path = "./?.lua;./?/init.lua"
-    return package.searchpath(name, path)
-end
+local function traverse_path(path)
+    path = fio.abspath(path)
+    local paths = { path }
 
-local function traverse_rocks(name, pathes_search)
-    local cwd = fio.cwd()
-    local index = string.len(cwd) + 1
-    local strerr = ""
-    while index ~= nil do
-        cwd = string.sub(cwd, 1, index - 1)
-        for i, path in ipairs(pathes_search) do
-            local file, err = package.searchpath(name, cwd .. path)
-            if err == nil then
-                return file
-            end
-            strerr = strerr .. err
-        end
-        index = string.find(cwd, "/[^/]*$")
+    while path ~= '/' do
+        path = fio.dirname(path)
+        table.insert(paths, path)
     end
-    return nil, strerr
-end
 
-local function search_rocks_lua(name)
-    local pathes_search = {
-        "/.rocks/share/tarantool/?.lua;",
-        "/.rocks/share/tarantool/?/init.lua;",
-    }
-    return traverse_rocks(name, pathes_search)
-end
+    return paths
+end
+
+-- Generate a search function, which performs searching through
+-- templates setup in options.
+--
+-- @param opts table with following parameters:
+--   - path_fn function which returns a base path for the
+--     resulting template
+--   - templates table with lua search templates
+--   - traverse bool flag which tells search function to build
+--     multiple paths by expanding base path up to the root ('/')
+-- @return a searcher function which builds a path template and
+-- calls package.searchpath
+local function gen_search_func(opts)
```

Comment:
Do you really need the options as a table? This is an internal function, so it is not necessary to pack arguments into a table for backward compatibility. Why not to make this function simpler?
```Lua
local function gen_search_func(path_fn, templates, need_traverse)
```
By the way, look at the new name of the flag - for flags we always add a verb like 'is_', 'need_', 'has_' etc.

Reply:
I usually prefer not to have too many function arguments. Also, named arguments are more readable from the caller point of view. But I don't insist. If this style is preferable in Tarantool code base, I'm ok to adjust.

Reply:
Switched to positional arguments

Reply:
From the callers point of view - maybe. From the maintenance and function readability part it is not good. When a function takes only one 'opts' parameter, you need to spend much more time on documenting each option. The comment pads out. Additionally, you can accidentally misspell an opt name, and you won't notice it in code, the expected key will be just nil. I've faced that problem already.